### PR TITLE
Cookie.newBuilder added.

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

File: okhttp/src/test/java/okhttp3/CookieTest.java [hunk 1]
```diff
@@ -423,6 +423,25 @@
...
+            .expiresAt(MAX_DATE)
+            .build();
+    assertThat(cookie.name()).isEqualTo("a");
+    assertThat(cookie.value()).isEqualTo("b");
+    assertThat(cookie.expiresAt()).isEqualTo(MAX_DATE);
+    assertThat(cookie.domain()).isEqualTo("example.com");
+    assertThat(cookie.path()).isEqualTo("/");
+    assertThat(cookie.secure()).isFalse();
+    assertThat(cookie.httpOnly()).isFalse();
+    // can't be unset
```
Comment (yschimke) [inline okhttp/src/test/java/okhttp3/CookieTest.java:440]: Worth discussing this, strictly speaking, this isn't worse than currently chaining these calls on a fresh Builder, but is not ideal either.

File: okhttp/src/main/kotlin/okhttp3/Cookie.kt [hunk 2]
```diff
@@ -262,6 +264,20 @@ class Cookie private constructor(
     private var persistent = false
     private var hostOnly = false
 
+    constructor()
+
+    internal constructor(cookie: Cookie) {
```
Comment (swankjesse) [inline okhttp/src/main/kotlin/okhttp3/Cookie.kt:269]: Can this be private?

File: okhttp/src/test/java/okhttp3/CookieTest.java (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/test/java/okhttp3/CookieTest.java:440]: 👍🏻

Comment (swankjesse) [reply okhttp/src/test/java/okhttp3/CookieTest.java:440]: That change is independent!

**Review by swankjesse — APPROVED** [review]

File: okhttp/src/main/kotlin/okhttp3/Cookie.kt (hunk 2 above)
Comment (yschimke) [reply okhttp/src/main/kotlin/okhttp3/Cookie.kt:269]: No, Cookie can't access Cookie.Builder(Cookie) if it is.

**Pull request merged**